src/lsm6.ts

```diff
@@ -0,0 +1,306 @@
+            }
+        });
+    }
+
+    public get accelerationG(): Vector3 {
+        return this._accel;
+    }
+
+    public get gyroDPS(): Vector3 {
+        return this._gyro;
+    }
+
+    public async enableDefault(): Promise<void> {
+        if (!this._isReady) {
+            return;
+        }
+
+        // Accelerometer
+        await this.setAccelerometerScale(AccelerometerScale.SCALE_2G);
+
+        // Gyro
+        await this.setGyroScale(GyroScale.SCALE_250_DPS);
+
+        // Common
+        await this._writeByte(RegAddr.CTRL3_C, IF_INC_ENABLED);
+    }
+
+    public async setAccelerometerScale(scale: AccelerometerScale): Promise<void> {
+        if (!this._isReady) {
+            return;
+        }
+
+        const controlByte = ACCEL_SCALE_CTRL_BYTE.get(scale);
+        await this._writeByte(RegAddr.CTRL1_XL, controlByte);
+        this._accelerometerScale = scale;
+
+        console.log(`[IMU] Accelerometer Scale: ${scale}`);
+    }
+
+    public async setGyroScale(scale: GyroScale): Promise<void> {
+        if (!this._isReady) {
+            return;
+        }
+
+        const controlByte = GYRO_SCALE_CTRL_BYTE.get(scale);
+        await this._writeByte(RegAddr.CTRL2_G, controlByte);
+        this._gyroScale = scale;
+
+        console.log(`[IMU] Gyro Scale: ${scale}`);
+    }
+
+    /**
+     * Read the current accelerometer values and update our internal data
+     *
+     * The values obtained from the raw I2C read operations are the
+     * 16-bit twos-complement raw IMU values. They will need to be
+     * multiplied by the sensitivity values to obtain a reading in mG.
+     *
+     * The final values that get stored are in G
+     *
+     * Linear Acceleration Sensitivity: LA_So (page 15)
+     */
+    public async readAccelerometer(): Promise<void> {
+        if (!this._isReady) {
+            return;
+        }
+
+        const xla = await this._readByte(RegAddr.OUTX_L_XL);
+        const xha = await this._readByte(RegAddr.OUTX_H_XL);
+        const yla = await this._readByte(RegAddr.OUTY_L_XL);
+        const yha = await this._readByte(RegAddr.OUTY_H_XL);
+        const zla = await this._readByte(RegAddr.OUTZ_L_XL);
+        const zha = await this._readByte(RegAddr.OUTZ_H_XL);
+
+        const accelX = (xha << 8) | xla;
+        const accelY = (yha << 8) | yla;
+        const accelZ = (zha << 8) | zla;
+
+        const tmpBuf = Buffer.alloc(2);
+
+        tmpBuf.writeUInt16BE(accelX, 0);
+        const accelXsigned = tmpBuf.readInt16BE(0);
+
+        tmpBuf.writeUInt16BE(accelY, 0);
+        const accelYsigned = tmpBuf.readInt16BE(0);
+
+        tmpBuf.writeUInt16BE(accelZ, 0);
+        const accelZsigned = tmpBuf.readInt16BE(0);
+
+        const scaleFactor = ACCEL_OUTPUT_SCALE_FACTOR.get(this._accelerometerScale);
+
+        this._accel.x = (scaleFactor * accelXsigned) / 1000;
+        this._accel.y = (scaleFactor * accelYsigned) / 1000;
+        this._accel.z = (scaleFactor * accelZsigned) / 1000;
```

Comment:
Why divide by 1000?

Reply:
The return values are in `mg` and `mdps`, so we divide by 1000 to get `g` and `dps`